### DISCUSSION TITLE
Events ingested, but not shown.

### DISCUSSION THREAD

**Discussion by blaggacao** [plausible/analytics#4127] [Self-Hosted Support] [answered]
Can you help me hint in the right direction?

In `events_v2`, I have:

```
[ ... ]
59 rows in set. Elapsed: 0.003 sec.

my.redacted.host :)
```

But the front-end keeps insisting:

![image](https://github.com/plausible/analytics/assets/7548295/41db492e-f2bf-4e30-b008-a9fedb9214f7)

Yet, also, somewhat more correctly:

![image](https://github.com/plausible/analytics/assets/7548295/ce3bbd36-db5d-4149-9ab1-d53c19e8ea2b)

**Comment by ruslandoga** [comment]
👋 @blaggacao 

Please share more information on your setup, `docker-compose.yml`, `plasuible-conf.env` with secrets redacted, etc. 

Also this thread might be helpful: https://github.com/plausible/analytics/discussions/3725

**Reply by blaggacao** [reply]
Thank you, your [suggested work arround here](https://github.com/plausible/analytics/discussions/3725#discussioncomment-8684428) did work.

**Reply by ruslandoga** [reply]
Just a quick check, do you have custom TZ env vars set in any of your containers? It's still unclear why this problem occurs.

**Reply by blaggacao** [reply]
I created the Site manually via the UI, I did _not_ use the sites api.

I believe there may be some issue in the code path which obviates the setting of the start date in some circumstances.

The service startup sequence is othewise:

```
/bin/plausible eval Plausible.Release.createdb
/bin/plausible eval Plausible.Release.migrate
/bin/plausible eval "(Plausible.Release.prepare() ; Plausible.Auth.create_user(\"$ADMIN_USER_NAME\", \"$ADMIN_USER_EMAIL\", \"$ADMIN_USER_PWD\"))"
/bin/psql -d plausible <<< "UPDATE users SET email_verified=true where email = '$ADMIN_USER_EMAIL';"
exec /bin/plausible start
```

Nothing extraordinary.

**Reply by blaggacao** [reply]
> Just a quick check, do you have custom TZ env vars set in any of your containers?

That could be the server (and runtime environment's) timezone is UTC+5

**Reply by blaggacao** [reply]
```
TZDIR=/nix/store/i6nk8llh46f2xjzc5h8j83kwwr1w3kx0-tzdata-2024a/share/zoneinfo
```

**Reply by blaggacao** [reply]
Ah!

```
native_stats_start_at
2024-05-23 08:36:52
```

That seemed to have been correctly set. However, not correctly read.

**Reply by blaggacao** [reply]
Maybe due to this line: https://github.com/plausible/analytics/blob/5881f1c1bf9fca587c669c8b33967c846d4ba2a4/lib/plausible/site/admin.ex#L27 :shrug:

**Reply by ruslandoga** [reply]
> That seemed to have been correctly set. However, not correctly read.

Is it supposed to be 2024-05-23 08:36:52 UTC+5?
What would be the correct way and what would be the incorrect way to read it?

> Maybe due to this line:

I don't think that line is relevant since CRM is disabled in the community edition.

---

My guess is it has something to do with non-UTC server timezone and this query https://github.com/plausible/analytics/blob/c81cb1693320b35a0d4a1fddbcb5d7bb7cbcf9bf/lib/plausible/stats/clickhouse.ex#L16-L20 which passes a "naive" (i.e. no tz info) sites.native_stats_start_at timestamp from PostgreSQL in a WHERE clause to ClickHouse. [sites.native_stats_start_at is set to `now()` on site insert.](https://github.com/plausible/analytics/blob/c81cb1693320b35a0d4a1fddbcb5d7bb7cbcf9bf/priv/repo/migrations/20230301095227_add_native_stats_start_date.exs#L14)

**Reply by blaggacao** [reply]
> Is it supposed to be 2024-05-23 08:36:52 UTC+5?
What would be the correct way and what would be the incorrect way to read it?

Now I'm confused.

```
id |      domain      |     inserted_at     |     updated_at      |    timezone    | public | locked | has_stats | imported_data | stats_start_date | ingest_rate_limit_scale_seconds | ingest_rate_limit_threshold | native_stats_start_at | domain_changed_from | domain_changed_at | allowed_event_props
----+------------------+---------------------+---------------------+----------------+--------+--------+-----------+---------------+------------------+---------------------------------+-----------------------------+-----------------------+---------------------+-------------------+---------------------
  1 | shop.exmaple.co | 2024-05-23 07:30:27 | 2024-05-23 08:36:52 | America/Bogota | f      | f      | f         |               | 2024-03-05       |                              60 |                             | 2024-05-23 08:36:52   |                     |                   |
```


I'm trying to remeber when I started the site:

- It was after the morning routine, that is it must have been **later** than 0900 UTC+2
- So if these values are UTC, then that would translate to creation at 0730UTC or 0930UTC+2 and updated (I tried to "restet stats" to see if it would solve my initial problem) at 0836 or 1036UTC+2 :heavy_check_mark: This is likely the case


So I conclude they are UTC values, indeed.

I'd have to learn elixir to be able to dig further, I guess :smile: 

I'm not sure if I'm being useful, here. Sry, if not.

**Comment by blaggacao** [comment, marked as answer]
Solved by https://github.com/plausible/analytics/discussions/3725#discussioncomment-8684428
